tools/calculate_build_cache_key.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+
+# This file is a helper script to calculate the hash for cache key
+# in the CI. It is not used in the build process.
+
+from hashlib import sha256
+from pathlib import Path
+
+import pathspec  # pip install pathspec
+
+
+def hash_file(filename):
+    with open(filename, "rb") as f:
+        return sha256(f.read()).hexdigest()
+
+
+def get_ignore_pattern(root: Path) -> list[str]:
+    ignorefile = root / ".gitignore"
+    if not ignorefile.exists():
+        raise RuntimeError("No .gitignore file found in root directory")
+
+    return ignorefile.read_text().splitlines()
+
+
+def main():
+    root: Path = Path(__file__).parent.parent
+    targets: list[Path] = [
+        root / "Makefile",
+        root / "Makefile.envs",
+        root / "pyodide-build",
+        root / "packages",
```

Comment:
This check is quite strict. Any changes in `pyodide-build` or `packages` will invalidate the cache. Maybe we can relax the condition in the future.